packages/components/src/styles/components/application-state.scss

```diff
@@ -7,43 +7,96 @@
 // APPLICATION-STATE COMPONENT
 //
 
-$hds-application-state-padding: 12px 0;
+$hds-application-state-content-max-width: 480px;
+$hds-application-state-footer-child-gap: 8px;
 
 .hds-application-state {
-  width: 19.5rem;
+  display: grid;
+  grid-template-columns: minmax(0, $hds-application-state-content-max-width) 1fr 1fr;
+  grid-template-rows: auto auto auto auto;
+  
   max-width: 100%;
   margin: 0 auto; // this will center the component in the parent container
+
+  &.hds-application-state--align-center {
+    grid-template-columns: 1fr minmax(0, $hds-application-state-content-max-width) 1fr;
+
+    .hds-application-state__media {
+      text-align: center;
+    }
+
+    .hds-application-state__header,
+    .hds-application-state__body,
+    .hds-application-state__footer {
+      grid-column: 2;
+    }
+  }
+}
+
+.hds-application-state__media {
+  grid-column: 1 / -1;
+  grid-row: 1;
+  
+  padding-bottom: 32px;
 }
 
 .hds-application-state__header {
+  grid-column: 1;
+  grid-row: 2;
+  
   display: grid;
   grid-template-columns: min-content 1fr;
+  grid-template-rows: auto auto;
   align-items: start;
-  color: var(--token-color-foreground-faint);
+  color: var(--token-color-foreground-strong);
+}
+
+.hds-application-state__error-code {
+  grid-column: 1 / -1;
 }
 
 .hds-application-state__icon {
+  grid-row: 2;
+  
   margin-right: 8px; // instead of column gap
   padding-top: 4px; // this seems to align the icon along with line 21
 }
 
-.hds-application-state__title,
-.hds-application-state__error-code {
-  grid-column-start: 2;
+.hds-application-state__title {
+  grid-row: 2;
+  
+  font-weight: var(--token-typography-font-weight-semibold);
+
+  &:not(.hds-application-state__icon + &) {
+    grid-column: 1 / -1;
+  }
 }
 
 .hds-application-state__body {
-  padding: $hds-application-state-padding;
-  color: var(--token-color-foreground-faint);
+  grid-column: 1;
+  grid-row: 3;
+  
+  padding: 12px 0;
+  color: var(--token-color-foreground-primary);
 }
 
 .hds-application-state__footer {
-  display: flex;
-  gap: 8px;
-  justify-content: space-between;
+  grid-column: 1;
+  grid-row: 4;
+
+  padding-top: 12px;
+
+  & > * {
+    float: left;
+    margin-right: $hds-application-state-footer-child-gap;
```

Comment:
I would use `display: flex` with `gap` for spacing vs margins. I don't think there's a reason to use float anymore for this type of layout in modern browsers which we support.

Float should be reserved for cases in which you want text content to actually flow around floated content such as an image. (Which was the original intention/purpose of float before it was co-opted for general content layout use before the advent of flexbox along with widespread browser support.)

Reply:
Flex doesn't work in this case specifically because we want the first 2 items on the left, and any remaining items on the right. This is the only way I could figure out how to make this work with dynamic content and without modifying the structure of the component. If you have a way you can make this work with flexbox, I would much rather use it. I also played around with a few implementations with css-grid to no avail.

Reply:
Followup: Zack and I discussed together over Slack and I recommended using margin-left: auto to push over the 3rd (and following) links to the right. ([Example CodePen](https://codepen.io/KristinB/pen/PorZgRE/7a6266aae97137a707e27bbc182c8b5d))